src/library/payfine/PayFineControl.java

```diff
@@ -0,0 +1,70 @@
+package library.payfine;
+import library.entities.Library;
+import library.entities.Patron;
+
+public class PayFineControl {
+    
+    private PayFineUI ui;
+    private enum ControlState { INITIALISED, READY, PAYING, COMPLETED, CANCELLED };
+    private ControlState state;
+    
+    private Library library;
+    private Patron patron;
+
+
+    public PayFineControl() {
+        this.library = Library.getInstance();
+        state = ControlState.INITIALISED;
+    }
+    
+    
+    public void setUI(PayFineUI ui) {
+        if (!state.equals(ControlState.INITIALISED)) {
+            throw new RuntimeException("PayFineControl: cannot call setUI except in INITIALISED state");
+        }    
+        this.ui = ui;
+        this.ui.SeTrEaDy();
+        state = ControlState.READY;
+    }
+
+
+    public void cardSwiped(long patronId) {
+        if (!state.equals(ControlState.READY))
```

Comment:
PR44 D1 CSG#7 Curly brackets Must be used

Reply:
[4f9a9bd](https://github.com/elliotmitchum/ITC205/pull/44/commits/4f9a9bd2d2709cfaafd9f3199a06a8325c12273b)